ubuntu-ruby-fips/Dockerfile

```diff
@@ -46,11 +85,86 @@ RUN apt-mark hold libssl1.1
 ## Copy postgres and ruby binaries from fips-artifacts
 RUN mkdir -p /var/lib/pgsql/ /var/lib/ruby/
 COPY --from=postgres-client-builder /usr/local/pgsql/ /usr/local/pgsql/
-COPY --from=ubuntu-ruby-builder /var/lib/ruby/ /var/lib/ruby/
+COPY --from=ubuntu-ruby-fips-builder /var/lib/ruby/ /var/lib/ruby/
 
 ENV PATH="/usr/local/pgsql/bin:/var/lib/ruby/bin:/usr/local/ssl/bin:${PATH}" \
+    DEBIAN_FRONTEND=noninteractive\
+    TZ=Etc/UTC \
+    BUNDLE_SILENCE_ROOT_WARNING=1 \
+    LANG=C.UTF-8 \
     LD_LIBRARY_PATH="/usr/local/ssl/lib" \
     OPENSSL_FIPS=1
 
-## Install Ruby tools
-RUN gem install --no-document bundler:$BUNDLER_VERSION
+# Conjur Base Image (Ubuntu) with dev tooling
+FROM ubuntu-ruby-fips-slim as ubuntu-ruby-fips-dev
+
+FROM golang:latest as setuser-builder
+WORKDIR /src
+COPY ./setuser /src
+RUN go build .
+
+# Conjur Base Image (Ubuntu) with postgres server
+FROM ubuntu:$UBUNTU_VERSION as ubuntu-ruby-postgres-fips
```

Comment:
This will be used as replacement for phusion-ruby-fips (only know location for me is in the appliance). There are more occurrences but mostly could be replaced with lighter ubuntu-ruby-builder just for build-essentials and ruby (for example for 'bundle install').
Actually our work targets to remove from conjur/appliance need of installing additional tools. In that case base images would take care of installing everything they need, and then in appliance there would be not install command.

Reply:
Hmm... okay. I'll let this be for now, then.

I'm not sure if I'm a fan of intentionally separating strongly coupled parts of the appliance where there isn't re-use sharing taking place. This goes against another effort we've been slowly undertaking to consolidate more of the appliance into a single repo, so that there's a single place to understand and change the appliance.

That's not something to resolve in this PR or project, though! 😅

Reply:
I agree with your concern @micahlee. I'd like to see us be very specific with the dependencies and where we apply them.  From the list below, I'd recommend we install `keyutils`, `slapd`, `ldap-utils`, `runit`, `libarchive13`, `cron`,  `syslog-ng-core`, and `syslog-ng-mod-sql` in the appliance as these are all dependencies specific to the appliance.  The rest of these dependencies are import for both Conjur and Conjur Enterprise, and thus, make sense to manage here.

Reply:
I assume that `ubuntu-ruby-postgres-fips` will be used only by appliance in it's Dockerfile.
We planned to deliver all needed dependencies in base-image repo and then in destination repo (for example Conjur or appliance) use that image only to configure and prepare application.